api/apis/integration/service_binding_test.go

```diff
@@ -0,0 +1,280 @@
+					"details":          HaveKeyWithValue("href", HaveSuffix("/v3/service_credential_bindings/"+guid+"/details")),
+				})))
+
+				serviceBinding := new(servicesv1alpha1.CFServiceBinding)
+				Expect(
+					k8sClient.Get(context.Background(), types.NamespacedName{Namespace: spaceGUID, Name: guid}, serviceBinding),
+				).To(Succeed())
+
+				Expect(serviceBinding.Labels).To(HaveKeyWithValue("servicebinding.io/provisioned-service", "true"))
+				Expect(serviceBinding.Spec.AppRef).To(Equal(corev1.LocalObjectReference{Name: appGUID}))
+				Expect(serviceBinding.Spec.Service).To(Equal(
+					corev1.ObjectReference{
+						APIVersion: "services.cloudfoundry.org/v1alpha1",
+						Kind:       "CFServiceInstance",
+						Name:       serviceInstanceGUID,
+					},
+				))
+			})
+		})
+	})
+
+	Describe("GET /v3/service_credential_bindings", func() {
+		const (
+			appGUID    = "the-app-guid"
+			secretName = "the-secret-name"
+		)
+
+		var (
+			serviceInstance1GUID, serviceBinding1GUID string
+			serviceInstance2GUID, serviceBinding2GUID string
+			spaceGUID                                 string
+		)
+
+		BeforeEach(func() {
+			org := createOrgAnchorAndNamespace(ctx, rootNamespace, generateGUID())
+			space := createSpaceAnchorAndNamespace(ctx, org.Name, "spacename-"+generateGUID())
+			spaceGUID = space.Name
+
+			cfApp := &workloadsv1alpha1.CFApp{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      appGUID,
+					Namespace: spaceGUID,
+				},
+				Spec: workloadsv1alpha1.CFAppSpec{
+					Name:         "name-for-" + appGUID,
+					DesiredState: "STOPPED",
+					Lifecycle: workloadsv1alpha1.Lifecycle{
+						Type: "buildpack",
+						Data: workloadsv1alpha1.LifecycleData{Buildpacks: []string{"java"}},
+					},
+				},
+			}
+			Expect(
+				k8sClient.Create(context.Background(), cfApp),
+			).To(Succeed())
+
+			serviceInstance1GUID = generateGUID()
+			serviceInstance1 := createServiceInstance(ctx, k8sClient, serviceInstance1GUID, spaceGUID, "service-instance-1-name", secretName)
+
+			serviceBinding1GUID = generateGUID()
+			serviceBindingName := "service-binding-1-name"
+			_ = createServiceBinding(ctx, k8sClient, serviceBinding1GUID, spaceGUID, &serviceBindingName, serviceInstance1.Name, cfApp.Name)
+
+			serviceInstance2GUID = generateGUID()
+			serviceInstance2 := createServiceInstance(ctx, k8sClient, serviceInstance2GUID, spaceGUID, "service-instance-2-name", secretName)
+
+			serviceBinding2GUID = generateGUID()
+			_ = createServiceBinding(ctx, k8sClient, serviceBinding2GUID, spaceGUID, nil, serviceInstance2.Name, cfApp.Name)
+
+			var err error
+			req, err = http.NewRequestWithContext(ctx, "GET", serverURI("/v3/service_credential_bindings"), strings.NewReader(""))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		JustBeforeEach(func() {
+			router.ServeHTTP(rr, req)
+		})
+
+		When("the user has permission to List ServiceBindings", func() {
```

Comment:
There is no test for the case when the user does not have permissions to list service bindings.